### PR TITLE
Replace Array by Vector or Matrix where appropriate

### PR DESCRIPTION
**Pull request by fingolfin**
This was done with the following commands:

    git ls-files | xargs perl -pi -e 's/Array\{([^{}]+(\{((?>[^{}]+)|(?2))*\})?),\s*2\s*\}/Matrix{\1}/g'
    git ls-files | xargs perl -pi -e 's/Array\{([^{}]+(\{((?>[^{}]+)|(?2))*\})?),\s*1\s*\}/Vector{\1}/g'
    git ls-files | xargs perl -pi -e 's/Array\{([^{}]+(\{((?>[^{}]+)|(?2))*\})?),\s*1\s*\}/Vector{\1}/g'

### CODE REVIEW

File: src/constructors.jl [hunk 1]
```diff
@@ -249,7 +249,7 @@ julia> val = GAP.evalstr("[ true, false, true ]")
 GAP: [ true, false, true ]
 
 julia> BitArray{1}(val)
-3-element BitArray{1}:
+3-element BitVector:
```
Comment (ThomasBreuer) [inline src/constructors.jl:252]: Strange: This is the only place where `BitArray{1}` got changed to `BitVector`. Why did the replacements find this one but not the others?

**Review by ThomasBreuer — CHANGES_REQUESTED** [review]
If I understand the changes right then the Julia output got adjusted after the regular expressions had been applied; otherwise I do not understand the changed spacing in `Dict` and `StepRange` outputs.
I think we should also replace `BitArray{1}` by `BitVector`, but this should then be all, according to the doctestfilters we use.

File: src/constructors.jl (hunk 1 above)
Comment (fingolfin) [reply src/constructors.jl:252]: Just to record in public what I mentioned in private chat: this second commit was made by running `doctest(GAP, fix=true)`.

**Comment by fingolfin** [comment]
I've added a commit which replace BitArray by BitVector

**Review by ThomasBreuer — APPROVED** [review]

**Pull request merged**